packages/office-ui-fabric-react/src/components/TeachingBubble/examples/TeachingBubble.Basic.Example.tsx

```diff
@@ -1,74 +1,39 @@
 import * as React from 'react';
-
 import { DefaultButton, IButtonProps } from 'office-ui-fabric-react/lib/Button';
 import { TeachingBubble } from 'office-ui-fabric-react/lib/TeachingBubble';
-
-export interface ITeachingBubbleBasicExampleState {
-  isTeachingBubbleVisible?: boolean;
-}
-
-export class TeachingBubbleBasicExample extends React.Component<{}, ITeachingBubbleBasicExampleState> {
-  private _menuButtonElement: HTMLElement;
-
-  constructor(props: {}) {
-    super(props);
-
-    this._onDismiss = this._onDismiss.bind(this);
-    this._onShow = this._onShow.bind(this);
-
-    this.state = {
-      isTeachingBubbleVisible: false,
-    };
-  }
-
-  public render(): JSX.Element {
-    const { isTeachingBubbleVisible } = this.state;
-    const examplePrimaryButton: IButtonProps = {
-      children: 'Try it out',
-    };
-    const exampleSecondaryButtonProps: IButtonProps = {
-      children: 'Maybe later',
-      onClick: this._onDismiss,
-    };
-
-    return (
-      <div className="ms-TeachingBubbleExample">
-        <span
-          className="ms-TeachingBubbleBasicExample-buttonArea"
-          ref={menuButton => (this._menuButtonElement = menuButton!)}
+import { useBoolean } from '@uifabric/react-hooks';
+
+const examplePrimaryButton: IButtonProps = {
+  children: 'Try it out',
+};
+
+export const TeachingBubbleBasicExample: React.FunctionComponent = () => {
+  const [teachingBubbleVisible, { toggle: toggleTeachingBubbleVisible }] = useBoolean(false);
+  const exampleSecondaryButtonProps: IButtonProps = {
```

Comment:
This is a good place for [useMemo](https://reactjs.org/docs/hooks-reference.html#usememo) (same in the other examples that use this pattern)
```suggestion
  const exampleSecondaryButtonProps: IButtonProps = React.useMemo(() => ({
    children: 'Maybe later',
    onClick: toggleTeachingBubbleVisible,
  }, [toggleTeachingBubbleVisible]);
```